rayon-core/src/registry.rs

```diff
@@ -716,3 +705,36 @@ pub fn in_worker<OP, R>(op: OP) -> R
         }
     }
 }
+
+/// [xorshift*] is a fast pseudorandom number generator which will
+/// even tolerate weak seeding, as long as it's not zero.
+///
+/// [xorshift*]: https://en.wikipedia.org/wiki/Xorshift#xorshift*
+struct XorShift64Star {
+    state: Cell<u64>,
+}
+
+impl XorShift64Star {
+    fn new() -> Self {
+        // Any non-zero seed will do -- this uses a stack address, which should be unique since
+        // we're only calling this once from each new thread.  The address is also shift-XORed to
+        // give us some high bits even on 32-bit platforms.
+        let base = 0;
+        let addr = &base as *const _ as u64;
```

Comment:
Hmm. I'm not crazy about this technique for getting a pointer into the stack. It feels fragile in the face of potential Rust unsafe code guidelines — e.g., I could sort of imagine `base` getting promoted to read-only memory or something. But probably we wouldn't be able to get away with such a rule anyway. It might be nice to have some kind of intrinsic for "random stack address", though — this is something I want to do from time to time. Anyhow I suppose it's fine "as is" for now, I filed https://github.com/nikomatsakis/rust-memory-model/issues/47 to track it and ponder it more deeply. =)